wicker/core/datasets.py

```diff
@@ -73,48 +242,27 @@ def __init__(
             defaults to FILE_LOCK_TIMEOUT_SECONDS
         :param treat_objects_as_bytes: If set, don't try to decode ObjectFields and keep them as binary data.
         """
-        super().__init__()
-        self._columns_to_load: Optional[List[str]] = columns_to_load
-        self._treat_objects_as_bytes = treat_objects_as_bytes
-        self._schema: Optional[DatasetSchema] = None
-        self._arrow_table: Optional[pyarrow.Table] = None
-
-        self._local_cache_path_prefix = local_cache_path_prefix
-        self._filelock_timeout_seconds = filelock_timeout_seconds
-        self._storage = storage if storage is not None else S3DataStorage()
-        self._s3_path_factory = s3_path_factory if s3_path_factory is not None else S3PathFactory()
-        self._column_bytes_file_cache = ColumnBytesFileCache(
-            local_cache_path_prefix=local_cache_path_prefix,
-            filelock_timeout_seconds=filelock_timeout_seconds,
-            path_factory=self._s3_path_factory,
-            storage=self._storage,
-            dataset_name=dataset_name,
-        )
-        self._pa_filesystem = (
+        pa_filesystem = (
             pafs.S3FileSystem(region=get_config().aws_s3_config.region) if pa_filesystem is None else pa_filesystem
         )
+        s3_path_factory = s3_path_factory if s3_path_factory is not None else S3PathFactory()
+        storage = storage if storage is not None else S3DataStorage()
 
-        self._dataset_id = DatasetID(name=dataset_name, version=dataset_version)
-        self._partition = DatasetPartition(dataset_id=self._dataset_id, partition=dataset_partition_name)
-        self._dataset_definition = DatasetDefinition(
-            self._dataset_id,
-            schema=self.schema(),
+        super().__init__(
+            dataset_name=dataset_name,
+            dataset_partition_name=dataset_partition_name,
+            dataset_version=dataset_version,
+            storage=storage,
```

Comment:
Same nit as before to re-sort the `storage` argument when passing to the base class `__init__`.

Reply:
Done